serrano/resources/preview.py

```diff
@@ -100,31 +105,34 @@ def get(self, request):
             iterable = processor.get_iterable(queryset=queryset,
                                               request=request)
 
-            exported = exporter.read(iterable,
-                                     request=request,
-                                     offset=offset,
-                                     limit=limit)
+            rows = exporter.manual_read(iterable,
+                                        request=request,
+                                        offset=offset,
+                                        limit=limit)
         else:
             iterable = processor.get_iterable(queryset=queryset,
                                               request=request,
                                               limit=limit,
                                               offset=offset)
 
-            exported = exporter.read(iterable, request=request)
+            # Select the requested reader
+            reader = params['reader']
 
-        for row in exported:
-            pk = None
-            values = []
+            if reader == 'threaded':
+                method = exporter.threaded_read
+            elif reader == 'cached':
+                method = exporter.cached_read
+            elif reader == 'cached_threaded':
+                method = exporter.cached_threaded_read
+            else:
+                method = exporter.read
```

Comment:
This is addressed in https://github.com/chop-dbhi/avocado/pull/315 Once merged, I will update this code to get the reader by called `exporter.reader` by name.